testing/sdk_marathon.py

```diff
@@ -52,19 +53,29 @@ def get_config(app_name, timeout=TIMEOUT_SECONDS):
     return config
 
 
-class _deployment_result(object):
-    def __init__(self, version, errmsg):
-        self._version = version
-        self._errmsg = errmsg
+class MarathonDeploymentResponse:
+    def __init__(self, response: requests.Response) -> None:
+        try:
+            response_json = response.json()
+        except ValueError:
+            log.error("Failed to parse marathon response as JSON: %s", response.text)
+            raise
+        log.info("Marathon deployment response JSON: %s", response_json)
+        response.raise_for_status()
```

Comment:
Should we `raise_for_status` *before* trying to decode the JSON?

Reply:
The idea behind it was to log the error message provided by Marathon before raising the potential exception. `response_json` includes during the error state a `message` field.